Tests/Runner/BranchedWorkspaceTest.php

```diff
@@ -0,0 +1,281 @@
+
+        $this->loadDataIntoTable($storageApi, $inBucketId, 'test-table', [
+            ['id', 'text'],
+            ['test1', 'test1'],
+        ]);
+
+        // create branch
+        $branchesApiMaster = new DevBranches($storageApiMaster);
+        $branchId = $this->cleanupAndCreateBranch($branchesApiMaster, self::BRANCH_NAME);
+        $storageApiWrapper = new ClientWrapper($storageApi, null, null, $branchId);
+
+        // setup configuration outside branch
+        $componentsApiForConfig = new Components($storageApi);
+        $configId = $this->generateConfigId();
+
+        $this->cleanupAndCreateConfiguration(
+            $componentsApiForConfig,
+            (new Configuration())
+                ->setComponentId('keboola.runner-workspace-test')
+                ->setName($configId)
+                ->setConfigurationId($configId)
+        );
+
+        // run testing job
+        $this->expectException(ClientException::class);
+        $this->expectExceptionMessage(sprintf('Configuration %s not found', $configId));
+
+        $this->runCopyJob($storageApiWrapper, $configId, $inBucketId, $outBucketId, self::TABLE_NAME);
+    }
+
+    public function testBranchConfigIsAvailableToBranchJob()
+    {
+        $storageApi = $this->getClient();
+        $storageApiMaster = $this->getMasterClient();
+
+        // prepare data
+        $inBucketId = $this->cleanupAndCreateBucket($storageApi, self::BUCKET_NAME, $storageApi::STAGE_IN);
+        $outBucketId = $this->cleanupAndCreateBucket($storageApi, self::BUCKET_NAME, $storageApi::STAGE_OUT);
+
+        $this->loadDataIntoTable($storageApi, $inBucketId, 'test-table', [
+            ['id', 'text'],
+            ['test1', 'test1'],
+        ]);
+
+        // create branch
+        $branchesApiMaster = new DevBranches($storageApiMaster);
+        $branchId = $this->cleanupAndCreateBranch($branchesApiMaster, self::BRANCH_NAME);
+        $storageApiWrapper = new ClientWrapper($storageApi, null, null, $branchId);
+
+        // setup configuration inside branch
+        $componentsApiForConfig = new Components($storageApiWrapper->getBranchClient());
+        $configId = $this->generateConfigId();
+
+        $this->createConfiguration(
+            $componentsApiForConfig,
+            (new Configuration())
+                ->setComponentId('keboola.runner-workspace-test')
+                ->setName('runner-tests')
+                ->setConfigurationId($configId)
+        );
+
+        // run testing job
+        $this->runCopyJob($storageApiWrapper, $configId, $inBucketId, $outBucketId, self::TABLE_NAME);
+    }
+
+    private function generateConfigId()
+    {
+        return uniqid('workspace-test-');
+    }
+
+    private function cleanupAndCreateBucket(Client $storageApi, $bucketName, $bucketStage)
+    {
+        try {
+            $bucketId = $storageApi->getBucketId('c-'. $bucketName, $bucketStage);
```

Comment:
nemyslim si, ze jsi takhle ziskal neco navic oproti
`$storageApi->dropBucket($bucketStage . '.c-' . $bucketName, ['force' => true]);`

Reply:
No, puvodni myslenka byla neresit vubec to skladani nazvu/ID bucketu - me je jedno, jaky vztah ma mezi sebou `name` a `id`, ze se to da nejak poskladat a k tomu se krasne hodila `getBucketId` metoda. Jenzepak jsem narazil na ten ojeb s prefixovanim `c-`, takze jsem to tu musel doplnit a uz to tak elegantne nevypada :(

Nicmene pokud by opravdu vzdy platilo, ze `bucketId = "{bucketStage}.c-{bucketName}"` (a ve vesech testech to opravdu tak je, co jsem koukal), proc to takle neni udelane primo v ty `getBucketId` metode a dela slozite listovani a hledani pres API?

Reply:
Na tuto otázku odpoví můj manwer Fanta Kocouwek -> #team-connection